dist_tables/querying.rst

```diff
@@ -120,3 +120,126 @@ Citus employs a two stage optimizer when planning SQL queries. The first phase i
 Citus’s distributed executor then takes these individual query fragments and sends them to worker PostgreSQL instances. There are several aspects of both the distributed planner and the executor which can be tuned in order to improve performance. When these individual query fragments are sent to the workers, the second phase of query optimization kicks in. The workers are simply running extended PostgreSQL servers and they apply PostgreSQL's standard planning and execution logic to run these fragment SQL queries. Therefore, any optimization that helps PostgreSQL also helps Citus. PostgreSQL by default comes with conservative resource settings; and therefore optimizing these configuration settings can improve query times significantly.
 
 We discuss the relevant performance tuning steps in the :ref:`performance_tuning` section of the documentation.
+
+Controlling Query Propagation
+#############################
+
+When the user issues a query, the Citus coordinator partitions it into smaller query fragments where each query fragment can be run independently on a worker shard. This allows Citus to distribute each query across the cluster.
+
+However the way queries are partitioned into fragments (and which queries are propagated at all) varies by the type of query. In some advanced situations it is useful to manually control this behavior. Citus provides utility functions to propagate SQL to workers, shards, or placements.
+
+Running on all Workers
+----------------------
+
+The least granular level of execution is broadcasting a statement for execution on all workers. This is useful for viewing a properties of entire worker databases or creating UDFs uniformly throughout the cluster. For example:
+
+.. code-block:: postgresql
+
+  -- Make a UDF available on all workers
+  SELECT run_command_on_workers($cmd$ CREATE FUNCTION ..... $cmd$);
+
+  -- List the size of each worker database
+  SELECT *
+    FROM run_command_on_workers($cmd$
+      SELECT pg_size_pretty(pg_database_size('citus'))
+    $cmd$);
+
+.. note::
+
+  The :code:`run_command_on_workers` function and other manual propagation commands in this section can run only queries which return a single column and single row.
+
+Running on all Shards
+---------------------
+
+Each distributed table :code:`foo` has shards across the workers which are each tables with a naming convention of :code:`foo_n` where :code:`n` is a number. In order to run a query against a shard, the query must be run on its containing worker, and the shard name must be interpolated into the query string as an identifier (:code:`%I`).
+
+.. code-block:: postgresql
+
+  -- Get the size of table shards to determine how balanced the cluster
```

Comment:
Are there better examples than table-size? We will have utility functions in 6.2 for distributed table size, so we may want to highlight a use-case other than that.